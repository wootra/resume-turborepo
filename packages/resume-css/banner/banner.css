.top-banner {
    text-align: center;
    min-height: 30px;
    background-color: rgb(241, 235, 216);
    padding: 0.2rem 0;
    margin-bottom: 1rem;
    position: sticky;
    top: 0;
    z-index: 1000;
    display: flex;
    flex-direction: column;
    justify-items: center;
    align-items: center;
}

.top-banner > * {
    display: flex;
    gap: 1rem;
    flex-direction: row;
    justify-items: center;
    align-items: center;
}

.top-banner select {
    padding: 0.3rem;
}

.top-banner select > option {
    text-align: center;
}
