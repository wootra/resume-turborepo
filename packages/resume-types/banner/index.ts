export type LinkMap = Readonly<{
    [key: string]: string;
}>;

export interface BannerProps {
    description: string;
    currentLabel: string;
    links: LinkMap;
}

export interface FrameworkNoticeBannerProps<T extends LinkMap> {
    description: string;
    currentLabel: keyof T;
}
