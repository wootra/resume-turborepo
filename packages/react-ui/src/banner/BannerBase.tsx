import { BannerProps } from 'resume-types';
import 'resume-css/banner/banner.css';
import 'resume-css/link/animate-underline.css';
import { bannerLinks } from 'common-data';

function onSelect(target: HTMLSelectElement) {
    const selected = target.value as keyof typeof bannerLinks;
    const selectedLink = bannerLinks[selected];
    window.location.href = selectedLink;
}

export const BannerBase = ({
    description,
    currentLabel,
    links,
    sourceLinks,
}: BannerProps) => {
    return (
        <div className='top-banner'>
            <div>
                <span>{description}</span>
                {links && currentLabel && (
                    <select
                        onChange={e => onSelect(e.currentTarget)}
                        value={currentLabel}
                    >
                        {Object.keys(links).map(label => (
                            <option
                                key={label}
                                // selected={currentLabel === label}
                                value={label}
                            >
                                {label}
                            </option>
                        ))}
                    </select>
                )}
            </div>
            {sourceLinks && (
                <div>
                    {Object.keys(sourceLinks).map(label => (
                        <a
                            key={label}
                            className='animate-underline'
                            href={sourceLinks[label]}
                        >
                            {label}
                        </a>
                    ))}
                </div>
            )}
        </div>
    );
};
