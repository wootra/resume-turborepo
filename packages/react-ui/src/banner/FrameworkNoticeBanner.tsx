import { FrameworkNoticeBannerProps } from 'resume-types';
import 'resume-css/banner/banner.css';
import { bannerLinks } from 'common-data';
import { BannerBase } from './BannerBase';

export const FrameworkNoticeBanner = ({
    description,
    currentLabel,
    sourceLinks,
}: FrameworkNoticeBannerProps<typeof bannerLinks>) => {
    return (
        <BannerBase
            description={description}
            currentLabel={currentLabel}
            links={bannerLinks}
            sourceLinks={sourceLinks}
        />
    );
};
