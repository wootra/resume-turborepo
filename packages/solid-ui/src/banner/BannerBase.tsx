import { BannerProps } from 'resume-types';
import 'resume-css/banner/banner.css';
import { Component, For } from 'solid-js';
import { bannerLinks } from 'common-data';

function onSelect(target: HTMLSelectElement) {
    const selected = target.value as keyof typeof bannerLinks;
    const selectedLink = bannerLinks[selected];
    window.location.href = selectedLink;
}

export const BannerBase: Component<BannerProps> = ({
    description,
    currentLabel,
    links,
}) => {
    return (
        <div class='top-banner'>
            <div>
                <span>{description}</span>
                <select onChange={e => onSelect(e.currentTarget)}>
                    <For each={Object.keys(links)}>
                        {label => (
                            <option
                                selected={currentLabel === label}
                                value={label}
                            >
                                {label}
                            </option>
                        )}
                    </For>
                </select>
            </div>
        </div>
    );
};
