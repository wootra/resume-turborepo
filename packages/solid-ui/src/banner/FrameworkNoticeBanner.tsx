import { FrameworkNoticeBannerProps } from 'resume-types';
import 'resume-css/banner/banner.css';
import { Component, For } from 'solid-js';
import { bannerLinks } from 'common-data';
import { BannerBase } from './BannerBase';

export const FrameworkNoticeBanner: Component<
    FrameworkNoticeBannerProps<typeof bannerLinks>
> = ({ description, currentLabel }) => {
    return (
        <BannerBase
            description={description}
            currentLabel={currentLabel}
            links={bannerLinks}
        />
    );
};
