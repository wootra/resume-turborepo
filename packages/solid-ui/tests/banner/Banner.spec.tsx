// @vitest-environment jsdom

import { FrameworkNoticeBanner } from '@/banner/FrameworkNoticeBanner';
import {
    describe,
    it,
    expect,
    beforeEach,
    afterEach,
    vi,
    SpyInstance,
} from 'vitest';
import { render, screen } from '@solidjs/testing-library';
// import {} from '@testing-library/user-event';
describe('FrameworkNoticeBanner', () => {
    let locationSpy: SpyInstance<[Location], void>;
    beforeEach(() => {
        locationSpy = vi.spyOn(window, 'location', 'set');
    });
    afterEach(() => {
        vi.restoreAllMocks();
    });

    describe('from vanilla to svelte', () => {
        let rendered: ReturnType<typeof render>;
        beforeEach(() => {
            rendered = render(() => (
                <FrameworkNoticeBanner
                    description='desc'
                    currentLabel='Vanilla JS'
                />
            ));
        });
        it('should work', () => {
            expect(rendered.container.innerHTML).toMatchSnapshot();
        });
    });

    describe('from svelte to vanilla', () => {
        let rendered: ReturnType<typeof render>;
        beforeEach(() => {
            rendered = render(() => (
                <FrameworkNoticeBanner
                    description='desc'
                    currentLabel='Astro JS(Svelte.js, Solid.js)'
                />
            ));
        });
        it('should work', () => {
            expect(rendered.container.innerHTML).toMatchSnapshot();
        });
    });
});
