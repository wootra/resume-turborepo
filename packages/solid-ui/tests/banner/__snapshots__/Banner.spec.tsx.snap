// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Banner > from svelte to vanilla > should work 1`] = `"<div class=\\"top-banner\\"><div><span>desc</span><select><option value=\\"Vanilla JS\\">Vanilla JS</option><option value=\\"Astro JS(Svelte.js, Solid.js)\\">Astro JS(Svelte.js, Solid.js)</option></select></div></div>"`;

exports[`Banner > from vanilla to svelte > should work 1`] = `"<div class=\\"top-banner\\"><div><span>desc</span><select><option value=\\"Vanilla JS\\">Vanilla JS</option><option value=\\"Astro JS(Svelte.js, Solid.js)\\">Astro JS(Svelte.js, Solid.js)</option></select></div></div>"`;
